url47e6aca666_pocorall_automaton_tgz-pJ8-net_pocorall_automaton_DatatypesJ8/src/net/pocorall/automaton/BasicAutomataFactory.java

```diff
@@ -0,0 +1,502 @@
+			b.append(c);
+			maxInteger(n, i + 1, b);
+		}
+		b.append(')');
+	}
+
+	/**
+	 * Constructs automaton that accept strings representing nonnegative integers
+	 * that are not less that the given value.
+	 * <p/>
+	 * bug found?? (pocorall)
+	 *
+	 * @param n string representation of minimum value
+	 */
+	public static DefaultAutomaton makeMinInteger(String n) {
+		int i = 0;
+		while (i + 1 < n.length() && n.charAt(i) == '0')
+			i++;
+		StringBuilder b = new StringBuilder();
+		b.append("0*");
+		minInteger(n.substring(i), 0, b);
+		b.append("[0-9]*");
+		return ((new RegExp(b.toString())).toAutomaton()).minimize();
+	}
+
+	private static void minInteger(String n, int i, StringBuilder b) {
+		b.append('(');
+		if (i < n.length()) {
+			char c = n.charAt(i);
+			if (c != '9')
+				b.append("[" + (char) (c + 1) + "-9][0-9]{" + (n.length() - i - 1) + "}|");
+			b.append(c);
+			minInteger(n, i + 1, b);
+		}
+		b.append(')');
+	}
+
+	/**
+	 * Constructs automaton that accept strings representing decimal numbers
+	 * that can be written with at most the given number of digits.
+	 * Surrounding whitespace is permitted.
+	 *
+	 * @param i max number of necessary digits
+	 */
+	public static DefaultAutomaton makeTotalDigits(int i) {
+		return ((new RegExp("[ \t\n\r]*[-+]?0*([0-9]{0," + i + "}|((([0-9]\\.*){0," + i + "})&@\\.@)0*)[ \t\n\r]*")).toAutomaton()).minimize();
+	}
+
+	/**
+	 * Constructs automaton that accept strings representing decimal numbers
+	 * that can be written with at most the given number of digits in the fraction part.
+	 * Surrounding whitespace is permitted.
+	 *
+	 * @param i max number of necessary fraction digits
+	 */
+	public static DefaultAutomaton makeFractionDigits(int i) {
+		return ((new RegExp("[ \t\n\r]*[-+]?[0-9]+(\\.[0-9]{0," + i + "}0*)?[ \t\n\r]*")).toAutomaton()).minimize();
+	}
+
+	/**
+	 * Constructs automaton that accept strings representing the given integer.
+	 * Surrounding whitespace is permitted.
+	 *
+	 * @param value string representation of integer
+	 */
+	public static DefaultAutomaton makeIntegerValue(String value) {
+		boolean minus = false;
+		int i = 0;
+		while (i < value.length()) {
+			char c = value.charAt(i);
+			if (c == '-')
+				minus = true;
+			if (c >= '1' && c <= '9')
+				break;
+			i++;
+		}
+		StringBuilder b = new StringBuilder();
+		b.append(value.substring(i));
+		if (b.length() == 0)
+			b.append("0");
+		DefaultAutomaton s;
+		if (minus)
+			s = makeChar('-');
+		else
+			s = makeChar('+').optional();
+		DefaultAutomaton ws = Datatypes.getWhitespaceAutomaton();
+		return (ws.concatenate(s.concatenate(makeChar('0').repeat()).concatenate(makeString(b.toString()))).concatenate(ws)).minimize();
+	}
+
+	/**
+	 * Constructs automaton that accept strings representing the given decimal number.
+	 * Surrounding whitespace is permitted.
+	 *
+	 * @param value string representation of decimal number
+	 */
+	public static DefaultAutomaton makeDecimalValue(String value) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.